lib/mergeDuplicateMaterials.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+var Cesium = require('cesium');
+var deepEqual = require('deep-equal');
+
+var defined = Cesium.defined;
+
+var removeUnusedMaterials = require('./removeUnusedMaterials');
+
+module.exports = mergeDuplicateMaterials;
+
+function mergeDuplicateMaterials(gltf) {
+    var materials = gltf.materials;
+    var materialIdMapping = {};
+    var uniqueMaterials = [];
+    for (var materialId in materials) {
+        if (materials.hasOwnProperty(materialId)) {
+            var material = materials[materialId];
+            var uniqueMaterialsLength = uniqueMaterials.length;
+            var unique = true;
+            for (var i = 0; i < uniqueMaterialsLength; i++) {
+                var uniqueMaterialId = uniqueMaterials[i];
+                var uniqueMaterial = materials[uniqueMaterialId];
+                if (deepEqual(material, uniqueMaterial)) {
+                    materialIdMapping[materialId] = uniqueMaterialId;
+                    unique = false;
+                    break;
+                }
+            }
+            if (unique) {
+                uniqueMaterials.push(materialId);
+            }
+        }
+    }
+    remapMaterials(gltf, materialIdMapping);
+    removeUnusedMaterials(gltf);
```

Comment:
Although it is not the best name, perhaps rename `mergeDuplicateMaterials` to `removeUnusedAndDuplicateMaterials`, then make `removeUnusedMaterials` private and do not call it from the main pipeline so it is not executed twice.  It kinda helps with performance, but the main reason is to keep the pipeline conceptually simple.

Same comment for all stages in this PR.